client/src/services/auth.js

```diff
@@ -10,6 +10,7 @@
 import { signInWithEmailAndPassword, signOut } from "firebase/auth";
 import FIREBASE_AUTH_ERROR_MESSAGES from "../constants/FIREBASE_AUTH_ERROR_MESSAGES";
 import { auth } from "../utils/firebase-config";
+import { sendData } from "./data";
```

Comment:
Is this function being used in this file? The linter says this is a dependency cycle, and it doesn't look like you ended up using `sendData` anywhere here - if not, then just remove this import to fix that dependency cycle